tests/suites/test_suite_ssl.function

```diff
@@ -3300,245 +3568,128 @@ exit:
-
-        TEST_ASSERT( mbedtls_ssl_conf_psk( &server.conf, psk_str->x,
-                             psk_str->len,
-                             (const unsigned char *) psk_identity,
-                             strlen( psk_identity ) ) == 0 );
+    perform_handshake( &options );
 
-        mbedtls_ssl_conf_psk_cb( &server.conf, psk_dummy_callback, NULL );
-    }
-#endif
-#if defined(MBEDTLS_SSL_RENEGOTIATION)
-    if( renegotiate )
-    {
-        mbedtls_ssl_conf_renegotiation( &(server.conf),
-                                        MBEDTLS_SSL_RENEGOTIATION_ENABLED );
-        mbedtls_ssl_conf_renegotiation( &(client.conf),
-                                        MBEDTLS_SSL_RENEGOTIATION_ENABLED );
+    /* The goto below is used to avoid an "unused label" warning.*/
+    goto exit;
+}
+/* END_CASE */
 
-        mbedtls_ssl_conf_legacy_renegotiation( &(server.conf),
-                                               legacy_renegotiation );
-        mbedtls_ssl_conf_legacy_renegotiation( &(client.conf),
-                                               legacy_renegotiation );
-    }
-#endif /* MBEDTLS_SSL_RENEGOTIATION */
+/* BEGIN_CASE depends_on:MBEDTLS_X509_CRT_PARSE_C:!MBEDTLS_USE_PSA_CRYPTO:MBEDTLS_PKCS1_V15:MBEDTLS_SSL_PROTO_TLS1_2 */
+void handshake_psk_cipher( char* cipher, int pk_alg, data_t *psk_str, int dtls )
+{
+    handshake_test_options options;
+    init_handshake_options( &options );
 
-    TEST_ASSERT( mbedtls_mock_socket_connect( &(client.socket),
-                                              &(server.socket),
-                                              BUFFSIZE ) == 0 );
+    options.cipher = cipher;
+    options.dtls = dtls;
+    options.psk_str = psk_str;
+    options.pk_alg = pk_alg;
 
-    TEST_ASSERT( mbedtls_move_handshake_to_state( &(client.ssl),
-                                                  &(server.ssl),
-                                                  MBEDTLS_SSL_HANDSHAKE_OVER )
-                 == 0 );
-    TEST_ASSERT( client.ssl.state == MBEDTLS_SSL_HANDSHAKE_OVER );
-    TEST_ASSERT( server.ssl.state == MBEDTLS_SSL_HANDSHAKE_OVER );
+    perform_handshake( &options );
 
-    if( cli_msg_len != 0 || srv_msg_len != 0 )
-    {
-        /* Start data exchanging test */
-        TEST_ASSERT( mbedtls_exchange_data( &(client.ssl), cli_msg_len,
-                                            expected_cli_fragments,
-                                            &(server.ssl), srv_msg_len,
-                                            expected_srv_fragments ) == 0 );
-    }
-#if defined(MBEDTLS_SSL_CONTEXT_SERIALIZATION)
-    if( serialize == 1 )
-    {
-        TEST_ASSERT( dtls == 1 );
-
-        TEST_ASSERT( mbedtls_ssl_context_save( &(server.ssl), NULL,
-                                               0, &context_buf_len )
-                                             == MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL );
+    /* The goto below is used to avoid an "unused label" warning.*/
+    goto exit;
```

Comment:
How do we know the handshake went well? Should we assert on `perform_handshake()` return value (assuming we add a return value to that function)?

Reply:
A global `test_info` structure is checked after each test case.